__tests__/Login.test.js

```diff
@@ -54,6 +60,19 @@ describe('Login Component', () => {
     expect(spy).toBeCalled();
   });
 
+  describe('checkAuthentication method', () => {
+    it('should call handleError method when the API call returns an error response', () => {
+      mock.onGet('/session').reply(400, {
+        error: { message: 'Bad request' },
+      });
+      const spy = createSpy('handleError');
+      wrapper.instance().checkAuthentication();
+      setImmediate(() => {
+        expect(spy).toBeCalled();
+      });
```

Comment:
when testing components, try as much as possible to test with respect to `state` and props and not the component's method.

Reply:
Thanks @Efosaok, this is also a unit test for the expected behavior of the function in a plain react app component.